### DISCUSSION TITLE
Copilot ads in the code viewer

### DISCUSSION THREAD

**Discussion by 0x3C50** [community/community#65073] [Copilot Conversations]
### Select Topic Area

General

### Body

![grafik](https://github.com/community/community/assets/99053360/4df3ec9b-39c2-450e-99aa-fafd04c29986)

Just discovered ads for copilot on the code viewer, and since the general feedback post seems to be deleted, I'll just share my thoughts about that here:

Ads like these are __absolutely unacceptable__, and even just the idea of putting ads into critical spaces like these should be immediately thrown out. I understand you want people to use your product, but ads like these are *not* how you achieve that. I thought I'd give it a try sometime soon, but this action single-handedly removed all of my incentive to do so.

**Comment by cschreib** [comment]
Came here to express the same thing. One of the main appeal of the GitHub web interface, to me, is that it is very lean and uncluttered. This message is distracting advertising, and has nothing to do with the content I am browsing. Please, please, do not do this, or at the very least add a "dismiss" button. This is a professional platform, treat it as such.

**Reply by 0x3C50** [reply]
Luckily, a "Do not show again" button does exist when clicking on the ad, but even still, this ad being there in the first place is disgusting.

**Reply by cschreib** [reply]
Thanks for the tip, although I am not about to click on an ad. They might accidentally think I was interested...

**Reply by joeweiss** [reply]
> They might accidentally think I was interested...

Exactly why I didn't see the "Do not show again" button. It's so weird to have an advertising prompt there. What's next, fake commit messages?

**Reply by guest271314** [reply]
The advertisement is an obvious lie. There is no way it is true everybody in all cases will code "55% faster" using their product. The individual won;t actually be coding at all. They'll be getting lazier by the second using the code somebody else wrote that the program produces. There is already "Try Copilot" advertisement in the profile drop down. Repugnant marketing. 

No thank you. I'll pass.

**Reply by guest271314** [reply]
Here you go.

One line of CSS to get rid of GitHub "Copiot" advertising on code view and preview pages on GitHub. Written by human hand, tested by human hand. No "artificial" intelligence involved. 

```
.react-code-size-details-in-header:nth-of-type(2),[data-testid="copilot-popover-button"]{display:none!important}
```

Browser extension,  tested on Chromium 118 and Firefox Nightly 119  [no-copilot-ad](https://github.com/guest271314/no-copilot-ad).

**Reply by thennothinghappened** [reply]
really doesn't belong sitting right there. that's ridiculous.

**Reply by Colonial-Dev** [reply]
Pretty convincing evidence that nobody is using Copilot long term, if they have to resort to such aggressive advertising tactics.

Not really surprising, IMO.

**Reply by fwcd** [reply]
@guest271314 `button[data-testid="copilot-popover-button"]` works too

**Reply by guest271314** [reply]
@fwcd GitHub is trying to be slick. The last time I checked the element is different in the non-preview mode of the repository cf. the preview mode, e.g., when you edit your code and click "Preview". That's why I used two (2) selectors.

**Comment by jl2** [comment]
Yeah, really disappointed when I noticed this.  It's easy enough to ad-block it, but I shouldn't have to on a service I'm paying for.

What bothers me the most is that the placement of the ad makes it look like *I'm* endorsing CoPilot.

**Comment by tgrandgent** [comment]
A similar problem exists in Outlook too:

https://answers.microsoft.com/en-us/outlook_com/forum/all/how-do-i-remove-the-copilot-ad-in-my-outlook-app/513730d0-90d1-4824-82eb-6888a9002e7e

**Reply by guest271314** [reply]
Spamming everything they can with that garbage.

**Comment by M7T5M3P** [comment]
Yeap quite disapointing got to admit it, but you can still use chat gpt

**Comment by Engineer-Hasibur-Rahman** [comment]
Yeah, it's pretty disappointing, got to admit it, but you can still use ChatGPT.

**Reply by guest271314** [reply]
I'm not interested in the garbage people call "artificial intelligence" at all. It's a scam.

**Comment by draeath** [comment]
Why is this advertisement still there? Get this garbage out of here.

If you use uBlock Origin, the following filter rule will remove this nonsense (at this time):

    github.com##[data-testid="copilot-popover-button"]
